pkg/extensions/tbtc/recovery/electrs.go

```diff
@@ -0,0 +1,77 @@
+package recovery
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"strings"
+)
+
+type httpClient interface {
+	Post(url string, contentType string, body io.Reader) (*http.Response, error)
+	Get(url string) (*http.Response, error)
+}
+
+// ElectrsConnection exposes a native API for interacting with an electrs http API.
+type ElectrsConnection struct {
+	apiURL string
+	client httpClient
+}
+
+// NewElectrsConnection is a constructor for ElectrsConnection.
+func NewElectrsConnection(apiURL string) *ElectrsConnection {
+	return &ElectrsConnection{
+		apiURL: apiURL,
+		client: http.DefaultClient,
+	}
+}
+
+func (e *ElectrsConnection) setClient(client httpClient) {
+	e.client = client
+}
+
+// Broadcast broadcasts a transaction the configured bitcoin network.
+func (e ElectrsConnection) Broadcast(transaction string) error {
+	if e.apiURL == "" {
+		for i := 0; i < 5; i++ {
+			logger.Warningf("Please broadcast Bitcoin transaction %s", transaction)
```

Comment:
Oh, I see the warning log has been moved here. I think it would be better if we pull it out to the previous location.

Reply:
I ended up keeping them in both places: if we configure the electrs URL to be `""` specifically, we do the 5x log, and if something goes wrong with the broadcast we fall back to the 5x log